### ISSUE TITLE
[Feature] Firebase Auth Integration

### ISSUE THREAD

**Issue by sharat87** [appsmithorg/appsmith#2207]
Perform authentication with the Firebase auth system.

**Comment by Johann-L-Pickard** [comment]
I would like to embed appsmith into an exiting react application, which uses firebase auth, I would like to prevent a double login situation. Currently, a jwt is generated in response to the username and password , I would like this same jwt for authorization into appsmith (which is embedded).

**Comment by Nikhil-Nandagopal** [comment]
@Johann-L-Pickard I assume you plan to embed appsmith as an iframe? If you're comfortable, you can pass the jwt generated to the iframe as a query param and read that inside appsmith. Let me know if you have another solution in mind

**Comment by Johann-L-Pickard** [comment]
@Nikhil-Nandagopal 
Yes, that is what I was thinking. Is that approach currently supported ? then appsmith would receive the token and authorize access without having to log in again ?

**Comment by mohanarpit** [comment]
@Johann-L-Pickard In order to do this, you will have to do the following: 
1. Deploy your Appsmith application as a public app. This will bypass all authentication requirements enforced by Appsmith.
2. Pass the JWT as a query parameter to the iframe that is rendering your embedded Appsmith application. Eg: `<appsmithURL>?jwtToken=abcd`
3. In the Appsmith application, you can read query params being sent to the page via: `{{appsmith.URL.queryParams.jwtToken}}`. You can bind the value of this variable to any API call that you are making to your system.
4. Your API is now responsible for verifying the integrity of the JWT token passed to it. If the token is null/invalid, your API can return an error. If it's valid, your API can return the requisite data. This data will be rendered by the Appsmith application. 

Do let me know if this works out for you. 

If you need any help in getting this setup, please shoot a DM on Discord to the `@appsmith-team` and we'll jump in to help resolve this.

**Comment by MrSaints** [comment]
> Pass the JWT as a query parameter to the iframe that is rendering your embedded Appsmith application. Eg: <appsmithURL>?jwtToken=abcd

Apologies for bumping a relatively stale issue.

Though this works, I wouldn't recommend it from a security point of view as such tokens should never be included in the URL as they may potentially be indexed in access logs, and leaked as a result. Even with a short expiration, I wouldn't suggest it.

Another potential option is using `postMessage` to communicate the token safely to the embedded app, and letting the app update the token using `storeValue` which it can then use for future API calls. For the record, I'm thinking out loud, and I haven't actually tried this.
